t/shared.t

```diff
@@ -33,5 +32,23 @@ is( $hash{multi}->{key1},      "foo", "multidimension, key1" );
 is( $hash{multi}->{arr1}->[0], "bar", "multidimension, arr1 - key0" );
 is( $hash{multi}->{arr1}->[1], "baz", "multidimension, arr1 - key1" );
 
-unlink("vars.db");
-unlink("vars.db.lock");
+{
+    @array = (0);
+    my @pids;
+    for my $i (0..5) {
+        my $pid = fork();
+        if ($pid == 0) {
+            # child
+            sleep 1;
```

Comment:
Something like this may work, even with fewer forks:
```
{
  @array = (0);
  my @pids;
  for my $i ( 0 .. 5 ) {
    my $pid = fork();
    if ( $pid == 0 ) {
      # child
      $SIG{CONT} = &push_and_exit;
      POSIX::pause;

      sub push_and_exit {
        push @array, 1;
        exit 0;
      }
    }
    push @pids, $pid;
  }

  print kill 'CONT' => @pids;
  waitpid $_, 0 for @pids;

  cmp_deeply \@array, [qw/0 1 1 1 1 1 1/], 'race condition avoided';
}
```
What do you think?